java/src/main/native/src/HostUDFWrapperJni.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cudf_jni_apis.hpp"
+
+#include <cudf/aggregation/host_udf.hpp>
+
+extern "C" {
+
+JNIEXPORT void JNICALL Java_ai_rapids_cudf_HostUDFWrapper_close(JNIEnv* env,
+                                                                jclass class_object,
+                                                                jlong ptr)
+{
+  try {
+    auto to_del = reinterpret_cast<cudf::host_udf_base*>(ptr);
+    delete to_del;
```

Comment:
👍

Reply:
Here we can colse all kind of derived UDF instances.

Reply:
Previously, it's in the derived classe, [link](https://github.com/NVIDIA/spark-rapids-jni/pull/2522/files#diff-49d015ba9b60e8c27bc2f98c09f63525807885bc6f6b1928af0a8ad1ba9ba2e8R63).